### ISSUE TITLE
Range of Similarity Logits (output of BNContrastiveHead)

### ISSUE THREAD

**Issue by pelinsuacar** [AILab-CVC/YOLO-World#401]
I observed that cls_logit is influenced by several factors in  BNContrastiveHead class. As a summary, after batch normalization the values of 'x' are typically centered around 0 with a standard deviation determined by the learned parameters. However, during training the parameters (batch norm scale and shift, logit scale and bias) will adapt and modify this range. When I check the output scores during inference like this: 

```python
data_info = runner.pipeline(dict(img_id=0, img_path=input_image))

    data_batch = dict(
        inputs=data_info["inputs"].unsqueeze(0),
        data_samples=[data_info["data_samples"]],
    )
    with autocast(enabled=False), torch.no_grad():
        output = runner.model.test_step(data_batch)[0]
        pred_instances = output.pred_instances
        
        print(pred_instances.scores)
``` 
I see that scores are always between 0 and 1. Is there another normalization step before that I miss or are these scores directly equal to the output of BNContrastiveHead class? @wondervictor Thank you for your time in advance !

```python
class BNContrastiveHead(BaseModule):
    """ Batch Norm Contrastive Head for YOLO-World
    using batch norm instead of l2-normalization
    Args:
        embed_dims (int): embed dim of text and image features
        norm_cfg (dict): normalization params
    """

    def __init__(self,
                 embed_dims: int,
                 norm_cfg: ConfigDict,
                 init_cfg: OptConfigType = None,
                 use_einsum: bool = True) -> None:

        super().__init__(init_cfg=init_cfg)
        self.norm = build_norm_layer(norm_cfg, embed_dims)[1]
        self.bias = nn.Parameter(torch.zeros([]))
        # use -1.0 is more stable
        self.logit_scale = nn.Parameter(-1.0 * torch.ones([]))
        self.use_einsum = use_einsum

    def forward(self, x: Tensor, w: Tensor) -> Tensor:
        """Forward function of contrastive learning."""
        x = self.norm(x)
        w = F.normalize(w, dim=-1, p=2)

        if self.use_einsum:
            x = torch.einsum('bchw,bkc->bkhw', x, w)
        else:
            batch, channel, height, width = x.shape
            _, k, _ = w.shape
            x = x.permute(0, 2, 3, 1)  # bchw->bhwc
            x = x.reshape(batch, -1, channel)  # bhwc->b(hw)c
            w = w.permute(0, 2, 1)  # bkc->bck
            x = torch.matmul(x, w)
            x = x.reshape(batch, height, width, k)
            x = x.permute(0, 3, 1, 2)

        x = x * self.logit_scale.exp() + self.bias
        return x
```
